receiver/kubeletstatsreceiver/kubelet/client.go

```diff
@@ -64,44 +54,90 @@ func newTLSClient(endpoint string, cfg *ClientConfig, logger *zap.Logger) (Clien
 	if err != nil {
 		return nil, err
 	}
-	return defaultTLSClient(endpoint, cfg.InsecureSkipVerify, rootCAs, clientCert, logger), nil
+	return defaultTLSClient(
+		endpoint,
+		cfg.InsecureSkipVerify,
+		rootCAs,
+		[]tls.Certificate{clientCert},
+		nil,
+		logger,
+	), nil
 }
 
-func defaultTLSClient(endpoint string, insecureSkipVerify bool, rootCAs *x509.CertPool, clientCert tls.Certificate, logger *zap.Logger) *tlsClient {
+func newServiceAccountClient(endpoint string, caCertPath string, tokenPath string, logger *zap.Logger) (*clientImpl, error) {
+	rootCAs, err := systemCertPoolPlusPath(caCertPath)
+	if err != nil {
+		return nil, err
+	}
+	tok, err := ioutil.ReadFile(tokenPath)
+	if err != nil {
+		return nil, err
+	}
+	tr := defaultTransport()
+	tr.TLSClientConfig = &tls.Config{
+		RootCAs: rootCAs,
+	}
+	return defaultTLSClient(endpoint, true, rootCAs, nil, tok, logger), nil
+}
+
+func defaultTLSClient(
+	endpoint string,
+	insecureSkipVerify bool,
+	rootCAs *x509.CertPool,
+	certificates []tls.Certificate,
+	tok []byte,
+	logger *zap.Logger,
+) *clientImpl {
 	tr := defaultTransport()
 	tr.TLSClientConfig = &tls.Config{
 		RootCAs:            rootCAs,
-		Certificates:       []tls.Certificate{clientCert},
+		Certificates:       certificates,
 		InsecureSkipVerify: insecureSkipVerify,
 	}
-	return &tlsClient{
+	if endpoint == "" {
+		endpoint = defaultEndpoint(logger)
+	}
+	return &clientImpl{
 		baseURL:    "https://" + endpoint,
 		httpClient: http.Client{Transport: tr},
+		tok:        tok,
 		logger:     logger,
 	}
 }
 
+func defaultEndpoint(logger *zap.Logger) (endpoint string) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		logger.Error("unable to get hostname", zap.Error(err))
+		endpoint = "localhost"
```

Comment:
Not sure if there is a case when this would work in k8s, but let's keep it if you think it can work

Reply:
Should work with hostNetwork, but I've removed falling back to localhost since getting the hostname shouldn't error out (and smart agent doesn't do this either).

Reply:
Yes, I was talking about localhost in this comment